include/CL/sycl/opencl_type.hpp

```diff
@@ -1,80 +1,69 @@
 #ifndef TRISYCL_SYCL_OPENCL_TYPE_HPP
 #define TRISYCL_SYCL_OPENCL_TYPE_HPP
 
-/** \file
-    triSYCL wrapper for OpenCL types
-
-    Joan DOT Thibault AT ens-rennes DOT fr
-
-    This file is distributed under the University of Illinois Open Source
-    License. See LICENSE.TXT for details.
-*/
-
-// Only provide OpenCL interoperability types when OpenCL is in use
-#ifdef TRISYCL_OPENCL
-
-#include <boost/compute.hpp>
+#include <boost/preprocessor/cat.hpp>
+#include <boost/preprocessor/comparison/equal.hpp>
+#include <boost/preprocessor/control/if.hpp>
+#include <boost/preprocessor/facilities/empty.hpp>
+#include <boost/preprocessor/list/for_each.hpp>
+#include <boost/preprocessor/logical/not.hpp>
+#include <boost/preprocessor/logical/or.hpp>
+#include <boost/preprocessor/seq/for_each.hpp>
+#include <boost/preprocessor/tuple/to_list.hpp>
+#include <boost/preprocessor/tuple/elem.hpp>
+
+#include "CL/sycl/vec.hpp"
 
 namespace cl {
 namespace sycl {
 
-/** Implement OpenCL interoperability types
-
-    \todo Implement all of them
-*/
-
-/** Wrapper of Boost::compute's cl_float3
- */
-class cl_float3 {
-
-  ::cl_float3 self;
-
-public :
-
-  cl_float3 () = default;
-
-
-  cl_float3 (::cl_float3 self_) : self { self_ }
-  {}
-
-
-  cl_float3 (float x, float y, float z) : self { x, y, z }
-  {}
-
-
-  /* Return the first element of the vector
-   */
-  auto& x() {
-    return self.s[0];
-  }
-
-
-  /* Return the second element of the vector
-   */
-  auto& y() {
-    return self.s[1];
-  }
-
-
-  /* Return the third element of the vector
-   */
-  auto& z() {
-    return self.s[2];
-  }
-
-};
-
-}
-}
-
-#endif
-
-/*
-    # Some Emacs stuff:
-    ### Local Variables:
-    ### ispell-local-dictionary: "american"
-    ### eval: (flyspell-prog-mode)
-    ### End:
-*/
-
-#endif //  TRISYCL_SYCL_OPENCL_TYPE_HPP
+#define SCALAR_TYPES                             \
```

Comment:
You should put all these preprocessor names in a more private "namespace" to limit collision probability with user code.
For example `TRISYCL__SCALAR_TYPES`